src/init.cpp

```diff
@@ -1150,8 +1150,9 @@ bool AppInitParameterInteraction()
     }
 
     fRequireStandard = !gArgs.GetBoolArg("-acceptnonstdtxn", !chainparams.RequireStandard());
-    if (chainparams.RequireStandard() && !fRequireStandard)
+    if (!chainparams.IsTestChain() && !fRequireStandard) {
```

Comment:
I didn't even know it was forbidden to use -acceptnonstdtxn=1 on mainchain...
Anyway, I don't like IsTestChain much, it is not descriptive. What about forbid_nonstd?
Or Leave this at RequireStandard and separate another DefaultRequireStandard or DefaultAcceptNonStd like @sipa suggested?

Reply:
I am happy to extend the docstring of `IsTestChain` if it is not descriptive enough, but I'd like to keep it unless others agree that it should go.

I think it is useful to have a single boolean param to indicate whether the chain's purpose is test-only.